CONTRIBUTING.md

```diff
@@ -19,3 +64,42 @@ Allowed types:
 - **test**: Adding missing tests or correcting existing tests
 
 _(based on [Angular contribution guidelines](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type))_
+
+#### Reference the GitHub issue
+
+To get a good paper trail and trace back any commits to their respective discussions, we reference each commit with the GitHub issue number that the commit is for. For that, put the issue number in parentheses at the end of the commit description:
+
+```
+<type>[optional scope]: <description> (#<issue number>)
+```
+
+[Example:](https://github.com/csshugs/FluidMS/commit/6d6fe3c362907c6c0e07896c7ba4785d5d0b3f1d)
+```
+fix: webserver not serving `index.html` on baseUrl (#13)
+```
+
+#### Give context to your commits
+
+To be able to identify the **WHY** behind a commit at a later point in time, we use the following template for our commit body:
+
+```
+<type>[optional scope]: <description> (#<issue number>)
+
+because:
+- [relevant context]
+- [why you decided to change things]
+- [reason you’re doing it now]
+
+this commit:
+- [does X]
+- [does Y]
+- [does Z]
+```
+
+— _[Source](https://twitter.com/r00k/status/1175100703829909505)_
+
+### Squash your commits
+
+When a pull request is approved, all the commits need to be [squashed](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing) in a meaningful manner before the pull request is merged. That is, combining those commits that form a complete feature into **one commit**. That doesn’t necessarily mean that all the commits of every pull request should ever result in one single commit! For example, if a pull request provides a new feature but alongside this new feature, a bug was fixed, too, this would result in two sqashed commits: One for the feature and one for the bugfix. The aim of this practice is to produce a sensible fragmentation of commits so we can later automatically create our changelog from these commits.
+
+If you feel unsure about this process, the FluidMS core team loves to do this job for you!
```

Comment:
consider adding on whats the prefered way (e.g. reach out in slack, request in PR, ...)